app/src/main/java/com/amaze/filemanager/adapters/AppsRecyclerAdapter.kt

```diff
@@ -445,15 +435,40 @@ class AppsRecyclerAdapter(
             fragment.context,
             CopyService::class.java
         )
-        val baseFile =
-            RootHelper.generateBaseFile(f, true)
+        val mainApkFile = RootHelper.generateBaseFile(baseApkFile, true)
         val startIndex = appDataParcelable.packageName.indexOf("_")
         val subString = appDataParcelable.packageName.substring(startIndex + 1)
-        baseFile.name = (appDataParcelable.label + "_$subString.apk")
-        ab.add(baseFile)
-        intent.putParcelableArrayListExtra(CopyService.TAG_COPY_SOURCES, ab)
+        val fileBaseName = appDataParcelable.label + "_$subString"
+        mainApkFile.name = "$fileBaseName.apk"
+        filesToCopyList.add(mainApkFile)
+        val splitPathList = appDataParcelable.splitPathList;
+        if (splitPathList != null) {
+            for (splitApkPath : String in splitPathList) {
+                val splitApkFile = File(splitApkPath)
+                val splitParcelableFile = RootHelper.generateBaseFile(splitApkFile, true)
+                var name = splitApkFile.name.lowercase()
+                if (name.endsWith(".apk")) {
+                    name = name.substring(0, name.length - 4)
+                }
+                val dotIdx = name.lastIndexOf('.')
+                name = name.substring(dotIdx + 1)
+                name = "${fileBaseName}_${name}.apk";
+                splitParcelableFile.name = name
+                filesToCopyList.add(splitParcelableFile)
+            }
+        }
+        intent.putParcelableArrayListExtra(CopyService.TAG_COPY_SOURCES, filesToCopyList)
         intent.putExtra(CopyService.TAG_COPY_TARGET, dst.path)
         intent.putExtra(CopyService.TAG_COPY_OPEN_MODE, 0)
+
+        val dest = Environment.getExternalStorageDirectory().path + "/app_backup"
+        Toast.makeText(
+                fragment.context,
+                fragment.getString(R.string.copyingapks, filesToCopyList.size, dest),
```

Comment:
Cant this just be `dst.path`?

Reply:
@VishnuSanal You are correct, the `dest` variable is no longer necessary. In the original code the toast was right at the beginning of the `popupBackup` method, at that point `dst` was not yet defined. I will update that.